x-pack/plugins/transform/public/app/common/validators/is_valid_frequency.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+
+import type { ParsedDuration } from './types';
+
+// only valid if value is up to 1 hour
+export function isValidFrequency(arg: unknown): arg is ParsedDuration {
```

Comment:
It would be useful to have a generic duration validator, e.g.
```
const getDurationValidator = ({  minDurationInMs: number; maxDurationInMs: number; allowedUnits: stiring[] })
```